nrfcloud/apiClient.ts

```diff
@@ -136,14 +188,28 @@ export const apiClient = ({
 				}).toString()}`,
 				{ headers },
 			)
-				.then<Page<Device>>(async (res) => res.json())
-				.then((devices) => ({ devices })),
+				.then<Static<typeof Devices>>(async (res) => res.json())
+				.then((devices) => {
+					const maybeDevices = validateWithTypeBox(Devices)(devices)
+					if ('errors' in maybeDevices) {
+						throw new Error(`Failed to validate response of 'listDevices' API`)
+					}
+
+					return { devices: maybeDevices.value }
+				}),
```

Comment:
Turn this into a reusable function, because these three changes in this file all have the same logic.

Do not swallow the validation errors, otherwise we don't know what part of the response was invalid.